capa/main.py

```diff
@@ -533,74 +611,23 @@ def get_extractor(
       UnsupportedOSError
     """
     if format_ not in (FORMAT_SC32, FORMAT_SC64):
-        if not is_supported_format(path):
-            raise UnsupportedFormatError()
-
-        if not is_supported_arch(path):
-            raise UnsupportedArchError()
-
-        if os_ == OS_AUTO and not is_supported_os(path):
-            raise UnsupportedOSError()
+        check_supported_format(path, os_)
```

Comment:
reconsidering, I don't feel like these additional functions necessarily make the code easier to read

especially here where it's not obvious that this may raise an exception...

Reply:
That is not something I thought about before, but this is helpful to consider. Thank you!